src/renderer/state.ts

```diff
@@ -329,41 +351,110 @@ export class AppState {
     );
     window.ElectronFiddle.addEventListener('before-quit', this.setIsQuitting);
 
+    /**
+     * Listens for changes in the app settings made in other windows
+     * and refreshes the current window settings accordingly.
+     */
+    window.addEventListener('storage', (event) => {
+      const key = event.key as GlobalSettingKey;
+      const { newValue } = event;
+
+      let parsedValue: unknown;
+
+      try {
+        parsedValue = JSON.parse(newValue as string) as unknown;
+      } catch {
+        // The new value is a plain string, not a well-formed stringified object.
+        parsedValue = newValue;
+      }
+
+      if (key in GlobalSetting && key in this) {
+        // Any settings listed here need morethan just updating the state directly.
+        const actions: Partial<Record<GlobalSettingKey, () => void>> = {
+          theme: () => {
+            this.setTheme(parsedValue as string);
+          },
+        };
+
+        const action = actions[key];
+
+        if (typeof action === 'function') {
+          action();
+        } else {
+          // Fall back to updating the state.
+          this[key] = parsedValue;
+        }
+      }
```

Comment:
Maybe if the key isn't recognized, a warning should be logged? Just in case something slips in without being added to the enum properly.

Reply:
🤔 There's another issue with keeping settings in sync between windows: I've also created a `WindowSpecificSetting` enum with settings meant to be window-specific rather than global (like the Electron version you want to use in that window). If I change the Electron version in the first window, the second window won't update its state right away because I only do that if the key is a member of `GlobalSetting`.

However, if the second window is reloaded for some reason, it will populate its state based on the localStorage and will pick up the Electron version that was set by the first window. Not sure if this is a common scenario, but we might need to ensure every window can save its own settings independently (maybe saving an object instead of a string and using the browserWindow ID as a key, like `localStorage.setItem(WindowSpecificSetting.version, JSON.stringify({ browserWindow1Id: '24.0.0', browserWindow2Id: '25.0.0' }))` or something like that). Do you think this should be addressed here?

Reply:
I think it's fine to ignore that scenario in this PR, seems reasonable enough that reloading a window (and thus losing all window-specific state) will default to the last Electron version set (whichever window it was set in). Trying to store window-specific settings in `localStorage` for a specific window seems overly complex for what is gained in practice.

If you want to go down that path in a follow-up PR, I'd suggest looking at using `sessionStorage` for `WindowSpecificSetting` - it will be specific to each window, and survive reloads. We'd probably want to layer it so that each window always sets the key in `sessionStorage`, as well as `localStorage`, and tries to read first from `sessionStorage` and then falls back to `localStorage` if it's not found (like on initial page load).